Prog/TOPA/TOPA_CBO_spreadsheet.sas

```diff
@@ -0,0 +1,104 @@
+**************************************************************************/
+
+%include "\\sas1\DCdata\SAS\Inc\StdLocal.sas";
+
+** Define libraries **;
+%DCData_lib( PresCat )
+
+%File_info( data=PresCat.TOPA_database, printobs=5 ) 
+%File_info( data=PresCat.TOPA_notices_sales, printobs=5 ) 
+
+/** Fill in missing columns from original TOPA database to notices sale database created**/
+data Topa_CBO_sheet; 
+  merge 
+    PresCat.TOPA_notices_sales (keep=id u_address_id_ref u_dedup_notice u_notice_date u_ownername u_sale_date  in=time_period)
+    Prescat.Topa_database (keep=id All_street_addresses Property_name Notes Technical_assistance_provider u_date_dhcd_received_ta_reg Tech_Assist_Staff Tenant_Assn_Lawyer Did_a_TA_claim_TOPA_rights TA_Development_Partner Date_TA_assignment_of_rights Developer_assignment_of_rights);
+  by id;
+  if time_period;
+  TA_negotiate=""; label TA_negotiate = "Did TA negotiate outside of TOPA process?";
+  ass_aff_developer=""; label ass_aff_developer ="Is Assignee an Affordable Housing developer?";
+  dev_agree=""; label dev_agree = "Development agreement? (Y/N)";
+  buyouts=""; label buyouts = "Number of buy outs"; 
+  assign_terms=""; label assign_terms = "Description of assignment terms";
+  add_notes=""; label add_notes = "Additional notes"; 
+run; 
+
+proc sort data=Topa_CBO_sheet out=Topa_CBO_sheet_sorted;
+  by u_address_id_ref u_notice_date;
+run;
+
+%File_info( data=Topa_CBO_sheet_sorted, printobs=20 ) 
+
+data Topa_CBO_sheet_retain; 
+  set Topa_CBO_sheet_sorted;
+  by u_address_id_ref u_notice_date;
+  format r_notes $char500.;
+  format r_TA_provider $char.;
+  format r_TA_staff $char.;
+  format r_TA_lawyer $char.;
+  format r_TA_claim_rights $char.;
+  format r_TA_dev_partner $char.;
+  format r_date_TA_ass_rigts $char.;
+  format r_dev_ass_right $char.;
+  if u_dedup_notice=0 then do;
+  r_notes=Notes; r_TA_provider=Technical_assistance_provider; r_TA_staff=Tech_Assist_Staff; r_TA_lawyer=Tenant_Assn_Lawyer; r_TA_claim_rights=Did_a_TA_claim_TOPA_rights;
+r_TA_dev_partner=TA_Development_Partner; r_date_TA_ass_rigts=Date_TA_assignment_of_rights; r_dev_ass_right=Developer_assignment_of_rights;
+end;
+  retain r_notes r_TA_provider r_TA_staff r_TA_lawyer r_TA_claim_rights r_TA_dev_partner r_date_TA_ass_rigts r_dev_ass_right; 
+  label r_notes ='Notes';
+  label r_TA_provider='CBO Technical assistance provider';
+  label r_TA_staff='Technical assistance staff';
+  label r_TA_lawyer='Technical assistance lawyer';
+  label r_TA_claim_rights='Did TA provider claim TOPA rights?';
+  label r_TA_dev_partner='TA Development Partner';
+  label r_date_TA_ass_rigts='Approx. Date of TA assignment of rights';
+  label r_dev_ass_right='Developer receiving assignment of rights';
+
+/*  if u_dedup_notice=1 then do;*/
+  if u_dedup_notice=1 and not( missing( Notes ) ) then r_notes=Notes;
+  else if u_dedup_notice=1 not( missing( Technical_assistance_provider ) ) then r_TA_provider=Technical_assistance_provider;
```

Comment:
You don't want to use ELSE IF here because the conditions are not mutually exclusive. That is, you can have missing values for Notes but not for Technical_assistance_provider. So they should all be independent IFs. 

Second, you left out the AND between the two conditions. `u_dedup_notice=1 AND not( missing( ...`  But, also, it is better to use the DO; ... END; block structure that you had earlier. If you are writing the same condition repeatedly (u_dedup_notice=1) then it is usually better to make that an outer IF and then have the other statements within a DO; ... END; block. That's the same as writing AND... repeatedly, but it is easier to read and more efficient.

Reply:
Some additional background on IF THEN ELSE. http://course1.winona.edu/thooks/Media/Handout%204%20-%20If-Then%20Statements%20in%20SAS.pdf